starfish/util/logging.py

```diff
@@ -0,0 +1,50 @@
+import os
+import platform
+from functools import lru_cache
+from json import JSONEncoder
+from subprocess import (
+    call,
+    check_output,
+    PIPE,
+    Popen,
+    STDOUT,
+)
+
+from starfish.types import CORE_DEPENDENCIES
+
+
+@lru_cache(maxsize=1)
+def get_core_dependency_info():
+    dependency_info = dict()
+    for dependency in CORE_DEPENDENCIES:
+        ps = Popen(('pip', 'show', dependency), stdout=PIPE)
```

Comment:
it is highly likely that [pkg_resources](https://setuptools.readthedocs.io/en/latest/pkg_resources.html) is a more portable way of determining this.